packages/did/src/DidDetails/FullDidDetails.ts

```diff
@@ -145,12 +158,33 @@ export async function authorizeTx(
     )
   }
 
+  if (!didDocument) {
+    didDocument = (await resolve(didUri)).didDocument as DidDocument
+  }
+  if (!didDocument?.id) {
```

Comment:
because it _should_ :) it's to shield against some dumbass passing in an empty object or something else entirely

Reply:
Mhh ok. The document is the result of calling the `resolve` function from the chain file, so I thought we are protected against those accidental cases? Otherwise basically nothing is considered safe?

Reply:
I would also be unpleasantly surprised if `resolve` doesn't protect against incorrect input.

Reply:
`resolve` is only called if you don't pass in the DidDocument yourself, but if you do, you can mess it up

Reply:
Oh now I get it. This is also a type guard for the initial input, like if the some random object is passed instead of a DID document.

Reply:
An even better type guard could perhaps be implemented, and could throw a specific error if an incorrect DID document format is provided.

Reply:
But maybe that's too much of an edge case 🤷

Reply:
I kind of like the idea of having a type guard that returns a valid `DidDocument`. Raphael what do you think? We could have one for a verification method, one for a service (I think we kinda have one already anyway), and combine them into one for a DID Document.

Reply:
that is a possibility! should this be part of this PR?

Reply:
I don't have a strong opinion about it. Feel free to mark this issue as resolved if you are not planning to put it in here and instead make a ticket for it.

Reply:
https://github.com/KILTprotocol/ticket/issues/2989